pkg/controller/workload/kubernetes/resource/resource.go

```diff
@@ -531,7 +530,6 @@ func setRemoteController(ctrl metav1.Object, obj metav1.Object) {
 	meta.AddAnnotations(obj, map[string]string{
 		RemoteControllerNamespace: ctrl.GetNamespace(),
 		RemoteControllerName:      ctrl.GetName(),
-		RemoteControllerUID:       string(ctrl.GetUID()),
```

Comment:
@hasheddan I was thinking this morning that perhaps it would be worth setting, but not checking the UID. This would give us some debuggability around which resource originally set the annotations. On the other hand it could be confusing that we set it but don't check it. What do you think? If you do elect to keep it we should probably add a comment near the namespace and name check explaining why we don't check it.